### PR TITLE
feat(device_info_plus): Add Swift Package Manager support

### PR DESCRIPTION
**Pull request by charafau**
ref: https://github.com/fluttercommunity/plus_plugins/issues/3152

Since SPM support is comming I plan to add it to all plugins. 

Followed the guide - https://docs.flutter.dev/packages-and-plugins/swift-package-manager/for-plugin-authors 
and checked it for both cocoapods and spm

### CODE REVIEW

File: packages/device_info_plus/device_info_plus/ios/device_info_plus/.build/workspace-state.json
Comment (miquelbeltran) [inline packages/device_info_plus/device_info_plus/ios/device_info_plus/.build/workspace-state.json:1]: Should this file be checked in?

File: packages/device_info_plus/device_info_plus/ios/device_info_plus/Sources/device_info_plus/include/device_info_plus/.gitkeep
Comment (miquelbeltran) [inline packages/device_info_plus/device_info_plus/ios/device_info_plus/Sources/device_info_plus/include/device_info_plus/.gitkeep:1]: I guess this file can be removed

File: packages/device_info_plus/device_info_plus/.gitignore [hunk 1]
```diff
@@ -45,3 +45,4 @@ build/
 .project
 .classpath
 .settings
+!.gitkeep
```
Comment (miquelbeltran) [inline packages/device_info_plus/device_info_plus/.gitignore:48]: Since .gitkeep would be removed, we don't need this change either

**Review by miquelbeltran — COMMENTED** [review]
I am not super familiar with the process yet, but took a look at it, and left some comments.

Also, we should do the same for the macOS part of the project, right? In that case, we should indicate in the Title that this only covers iOS

File: packages/device_info_plus/device_info_plus/ios/device_info_plus/.build/workspace-state.json
Comment (koji-1009) [reply packages/device_info_plus/device_info_plus/ios/device_info_plus/.build/workspace-state.json:1]: I suggest updating the root `.gitignore` to remove two directory from all packages.
The following PR is helpful.

https://github.com/flutter/packages/pull/6705

**Comment by charafau** [comment]
> I am not super familiar with the process yet, but took a look at it, and left some comments.
> 
> Also, we should do the same for the macOS part of the project, right? In that case, we should indicate in the Title that this only covers iOS

Thanks for fast review, this is also my first time doing it. I will apply suggestions from you and @koji-1009 and upload today. Hopefully migrate all plugins

**Comment by charafau** [comment]
Thanks guys, should be good now. I also added support for macos

**Comment by miquelbeltran** [comment]
Please take a look at the macos integration jobs, as they are now failing.

**Comment by charafau** [comment]
> Please take a look at the macos CI jobs, as they are now failing.

Sorry didn't notice that it's a scrollable list 😅  fixed and is passing now

File: packages/device_info_plus/device_info_plus/ios/device_info_plus.podspec [hunk 2]
```diff
@@ -14,11 +14,11 @@ Downloaded by pub (not CocoaPods).
   s.author           = { 'Flutter Community Team' => 'authors@fluttercommunity.dev' }
   s.source           = { :http => 'https://github.com/fluttercommunity/plus_plugins/tree/main/packages/device_info_plus' }
   s.documentation_url = 'https://pub.dev/packages/device_info_plus'
-  s.source_files = 'Classes/**/*'
-  s.public_header_files = 'Classes/**/*.h'
+  s.source_files = 'device_info_plus/Sources/device_info_plus/**/*.{h,m}'
+  s.public_header_files = 'device_info_plus/Sources/device_info_plus/include/**/*.h'
```
Comment (loic-sharma) [inline packages/device_info_plus/device_info_plus/ios/device_info_plus.podspec:18]:
```suggestion
  s.public_header_files = 'device_info_plus/Sources/device_info_plus/include/**/*.h'
```

File: packages/device_info_plus/device_info_plus/ios/device_info_plus/Package.swift [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+            dependencies: [],
+            resources: [
+                .process("PrivacyInfo.xcprivacy"),
+            ],
+            cSettings: [
+                .headerSearchPath("include/device_info_plus")
+            ]
+        )
+    ]
+)
```
Comment (loic-sharma) [inline packages/device_info_plus/device_info_plus/ios/device_info_plus/Package.swift:27]:
```suggestion
)

```

**Review by vbuberen — APPROVED** [review]
Tested on iOS and MacOS devices with both SPM on/off and everything works alright. 

Thanks for your contribution. Merging.

**Pull request merged**